src/libopensc/cwa-dnie.c

```diff
@@ -402,6 +605,59 @@ static int dnie_read_certificate(sc_card_t * card, char *certpath, X509 ** cert)
 	LOG_FUNC_RETURN(card->ctx, res);
 }
 
+/**
+ * Method that sets the configuration channel data to use.
+ * The configuration data is already set to the card private data.
+ * Just created in case this will be modified.
+ *
+ * @param card Pointer to card driver structure
+ * @param data The data for the channel will be assined here
+ * @return SC_SUCCESS if ok; else error code
+ */
+static int dnie_get_channel_data(sc_card_t * card, dnie_channel_data_t ** data) {
+	dnie_private_data_t *priv_data = GET_DNIE_PRIV_DATA(card);
+	LOG_FUNC_CALLED(card->ctx);
+	if (!priv_data->channel_data) {
+		sc_log(card->ctx, "Data channel configuration was not initialized");
+		LOG_FUNC_RETURN(card->ctx, SC_ERROR_INTERNAL);
+	}
+	*data = priv_data->channel_data;
+	LOG_FUNC_RETURN(card->ctx, SC_SUCCESS);
+}
+
+/**
+ * Method to assign into the private data the secure channel
+ * configuration to use. Right now the icc_intermediate_ca_cert
+ * issuer is used. If it is the new one the new data is assigned
+ * else the old data is set.
+ *
+ * @param card Pointer to card driver structure
+ * @param icc_intermediate_ca_cert Pointer to the X509 icc intermediate CA certificate
+ * @return SC_SUCCESS if ok; else error code
+ */
+static int dnie_set_channel_data(sc_card_t * card, X509 * icc_intermediate_ca_cert) {
+	char *buf = NULL;
+	dnie_private_data_t *priv_data = GET_DNIE_PRIV_DATA(card);
+	LOG_FUNC_CALLED(card->ctx);
+
+	X509_NAME *issuer = X509_get_issuer_name(icc_intermediate_ca_cert);
+	buf = X509_NAME_oneline(issuer, buf, 0);
```

Comment:
Please check if OpenSSL requires you to supply a valid issuer (i.e. != NULL)

Reply:
NULL is checked now just in case.